File: src/main/java/ro/alina/unidoc/model/StudentDocumentRowModel.java
package ro.alina.unidoc.model;

import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Builder
@Data
@Getter
@Setter
public class StudentDocumentRowModel {
    private final StudentModel studentModel;
    private final Long documentId;
    private final String studyGroup;
    private final Long studyGroupId;
    private final String name;
    private final String description;
    private final String filePath;
    private final LocalDateTime dateOfUpload;
    private final String status;
    private final Long notificationId;
    private final String responseDocumentFilePath;
}

File: src/app/model/student-document-row.model.ts
import {StudentModel} from "./student.model";

export interface StudentDocumentRowModel{
  documentId: number;
  studentModel: StudentModel;
  studyGroup: string;
  studyGroupId: number;
  name: string;
  description: string;
  filePath: string;
  dateOfUpload: string;
  status: string
  notificationId: number;
  responseDocumentFilePath: string;
}
